sound/soc/sof/intel/hda.c

```diff
@@ -992,6 +992,9 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 				dmic_num = hda_dmic_num;
 
 			switch (dmic_num) {
+			case 1:
+				dmic_str = "-1ch";
+				break;
```

Comment:
@juimonen can you provide a link to the GitHub/bugzilla where such devices are listed?

Does SOF actually support true mono configurations? I don't recall if anyone has ever tested this, e.g. the topology assumes 2ch volume controls?

Also if this is indeed the case what we can have a single mic, how do we know which PDM interface it's connected to? Are we making the assumption that channel 0 and 1 are on PDM0?

And last, if we can have 1 ch, should we plan for the case where there are 3 mics?

Reply:
@plbossart, this was customer request and yes it was indeed a device with 1 dmic.

The idea was as follow: @singalsu told that 2ch topology should work with single dmic (making it dual mono), and we have currently some chromebook with 1 dmic. However, it is not using the hda-generic topologies.

So we could probably just fall through with 1 dmic to 2ch topology (name), but @singalsu pointed out that then we kind of loose the future possibility to separate true mono topology. And to your question, small tests that @singalsu did today with true mono setup seems to indicate, that true mono doesn't just work out of the box very easily. Hence, I first suggest we just copy the hda-generic-2ch -> hda-generic-1ch -> with 1 dmic we would get dual mono to user space (hopefully, to be tested better). Later on if we get the true mono working, or someone even wants it, we have the possibility to update the topology.

I guess the assumption is that channel 0 and 1 are on PDM0, need to see what is the best solution, can we have it easily configured differently? I don't have access to customers device atm.

Yes we should probably prepare for arbitrary number of dmics, however I have difficulties to vision 3 dmics. OTH we didn't envisage 1 dmic either. 

<nonsense>
I remember a story where US presidents we're talking to mic setup consisting of 3 shure sm57 or 58 taped together. When the sound engineer was asked the reason for this odd spatial array setup the answer was "well you don't want to screw things up when the president is talking, if 1 brakes you still have 2 spares."
</nonsense>

Reply:
ok @juimonen let's add support for 1 and 3 channels, generate the topology files from the 2 and 4 ch cases and think later.